cg/models/demultiplex/demux_results.py

```diff
@@ -84,11 +126,22 @@ def barcode_report(self) -> Path:
     @property
     def demux_sample_sheet_path(self) -> Path:
         """Return the path to sample sheet in demuxed flowcell dir"""
+        demux_sample_sheet_path = {
+            "bcl2fastq": self.bcl2fastq_demux_sheet_path,
+            "dragen": self.dragen_demux_sheet_path,
```

Comment:
this could be a good case for using class inheritance!
Of course feel free to prioritize your time as you see fit
but for instance, one could create a base class for DemuxResults, and have classes DemuxNode(DemuxResults) and DemuxDagen(Demuxresults) have functions with same name which override the dafault behavior. That way you can always use the same methods for this class, but method will act slightly different depending on what type on DemuxResults is being used